### PR TITLE
Add "profile" loglevel

### PR DESCRIPTION
**Pull request by terminalmage**
This new loglevel sits between INFO and DEBUG, so that logging can be added to
aid in troubleshooting, without having to sift through all the debug logging.

This also adds a decorator which can be used on any function which will log
execution time, and also adds profile logging for template rendering.

### CODE REVIEW

File: salt/utils/decorators/__init__.py [hunk 1]
```diff
@@ -140,6 +142,29 @@ class depends(Depends):  # pylint: disable=C0103
     '''
 
 
+def timing(function):
+    '''
+    Decorator wrapper to log execution time, for profiling purposes
+    '''
+    def wrapped(*args, **kwargs):
```
Comment (s0undt3ch) [inline salt/utils/decorators/__init__.py]: You should add a `functions.wraps(function)` decorator to `wrapped` to keep the `wrapped` function attributes equal to those of `function`...

File: salt/utils/decorators/__init__.py [hunk 2]
```diff
@@ -1,18 +1,20 @@
 # -*- coding: utf-8 -*-
```
Comment (s0undt3ch) [inline salt/utils/decorators/__init__.py]: You changed the file perms to 755

**Comment by s0undt3ch** [comment]
:awesome:

File: salt/utils/decorators/__init__.py (hunk 2 above)
Comment (terminalmage) [inline salt/utils/decorators/__init__.py]: That was unintentional, thanks for catching it. I had backed up my git clones to a flash drive when my old work laptop was in the process of dying, and the flash drive was formatted FAT32, so this messed up the file permissions when I copied them back onto the EXT4 partition on my new work laptop.

**Comment by terminalmage** [comment]
@s0undt3ch I fixed the perms and rebased that fix into the original commit to ``salt/decorators/__init__.py``. I also added the decorator. Can you have another look? Thanks for the review.

**Comment by s0undt3ch** [comment]
:+1:

**Pull request merged**